### PR TITLE
Custom Copy Text Component

### PR DESCRIPTION
**Pull request by ymekuria**
Description 
Addresses #69 

This PR adds a custom copy text component that allows a user to copy code snippets with line numbers and ellipses removed.

### CODE REVIEW

File: src/theme/CodeBlock/CopyButton/index.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import clsx from 'clsx';
+// @ts-expect-error: TODO, we need to make theme-classic have type: module
+import copy from 'copy-text-to-clipboard';
+import { translate } from '@docusaurus/Translate';
+import styles from './styles.module.css';
+export default function CopyButton({ code, className }) {
+  const [isCopied, setIsCopied] = useState(false);
+  const copyTimeout = useRef(undefined);
+  const handleCopyCode = useCallback(() => {
+    const removeLineNumbers = /^\s*\d+\s/gm;
```
Comment (jasongitmail) [inline src/theme/CodeBlock/CopyButton/index.js:11]: Do the line numbers always have one or more spaces preceding the digits?

**Review by jasongitmail — APPROVED** [review]
Works great! Tested it on the two tutorials. This will be really helpful for people!

**Comment by jasongitmail** [comment]
@ymekuria Merging this because it worked in my testing and is super valuable. When you get a chance, curious about the question I left above still.

File: src/theme/CodeBlock/CopyButton/index.js (hunk 1 above)
Comment (ymekuria) [reply src/theme/CodeBlock/CopyButton/index.js:11]: @jasongitmail this is always the case with the current formatting of the code snippets with line numbers on all the tutorials. When there are single and double digit line numbers on the same code snippet, Double digit numbers have 1 preceding space and single digit numbers have 2 preceding spaces. I assume the spacing will be similar when there are double digit and triple digit numbers on the same code snippet. I can create another quick pr to handle all edge cases when there possibly are no preceding spaces.

**Pull request merged**